Assets/SEE/UI/PlatformDependentComponent.cs

```diff
@@ -166,9 +310,92 @@ protected virtual void UpdateTouchGamepad() { }
+        /// </summary>
+        [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+        protected class ManagedUIAttribute : Attribute
+        {
+            /// <summary>
+            /// Whether the object this attribute is on has a composition relation to the
+            /// <see cref="PlatformDependentComponent"/> it is contained in, that is, whether this object should be
+            /// destroyed when the <see cref="PlatformDependentComponent"/> is destroyed.
+            /// </summary>
+            public readonly bool Destroy;
+
+            /// <summary>
+            /// Whether this <see cref="PlatformDependentComponent"/> also has a composition relation
+            /// to the UI object this attribute is on, that is, whether this <see cref="PlatformDependentComponent"/>
+            /// should be destroyed when the object this attribute is on is destroyed.
+            /// </summary>
+            /// <remarks>
+            /// If this is set to <c>true</c>, we will need to check for destruction of the managed UI objects
+            /// in every <see cref="Update"/> frame, which may be costly. For this reason, we cache the integral UI objects
+            /// on <see cref="Start"/>, meaning that <em>they must not be re-assigned outside of <see cref="Start"/>!</em>
+            /// </remarks>
+            public readonly bool Integral;
+
+            /// <summary>
+            /// Whether the object this attribute is on should be enabled/disabled when the
+            /// surrounding <see cref="PlatformDependentComponent"/> is enabled/disabled.
+            /// </summary>
+            public readonly bool ToggleEnabled;
+
+            /// <summary>
+            /// Marks a Unity <see cref="Object"/> or enumerable of such objects as managed by the
+            /// <see cref="PlatformDependentComponent"/> it is contained in.
+            /// </summary>
+            /// <param name="destroy">Whether the object should be destroyed when the <see cref="PlatformDependentComponent"/>
+            /// is destroyed.</param>
+            /// <param name="integral">Whether the <see cref="PlatformDependentComponent"/> should be destroyed when
+            /// the object this attribute is on is destroyed.</param>
+            /// <param name="toggleEnabled">Whether the object should be enabled/disabled when the
+            /// <see cref="PlatformDependentComponent"/> is enabled/disabled.</param>
+            /// <remarks>
+            /// Note that, if <paramref name="integral"/> is set to <c>true</c>, the corresponding GameObject must not
+            /// be re-assigned outside of the <see cref="Start"/> method.
+            /// </remarks>
+            public ManagedUIAttribute(bool destroy = true, bool integral = false, bool toggleEnabled = false)
+            {
+                Integral = integral;
+                Destroy = destroy;
+                // TODO: Should the default value for ToggleEnabled be true or false?
```

Comment:
This is actually a question for reviewers: Should the default be that managed UI objects are automatically enabled/disabled along with the PlatformDependentComponent that contains them?

Reply:
What were your thoughts behind setting the default to false?

Reply:
This was mostly just done out of caution. There are a lot of PlatformDependentComponents by now, and I'm not confident that each one will work correctly if we enable this. On the other hand, I'm also not sure why it *wouldn't* work, but I rather wanted to wait for other opinions before making this change.

Reply:
Yeah, the question is which type of component is more common. Maybe I'm overthinking it. It's probably best to just leave it as false for now, it can after all always be set to true when that behavior is desired.